Remora.Discord.Commands/Extensions/CommandTreeExtensions.cs

```diff
@@ -206,30 +206,56 @@ public static Result<IReadOnlyList<IBulkApplicationCommandData>> CreateApplicati
 
             // Translate from options to bulk data
             Optional<ApplicationCommandType> commandType = default;
-            Optional<bool> defaultPermission = default;
+            Optional<bool?> directMessagePermission = default;
+            Optional<IDiscordPermissionSet?> defaultMemberPermissions = default;
             switch (node)
             {
                 case GroupNode groupNode:
                 {
-                    var defaultPermissionAttributes = groupNode.GroupTypes.Select
+                    var memberPermissionAttributes = groupNode.GroupTypes.Select
                         (
-                            t => t.GetCustomAttribute<DiscordDefaultPermissionAttribute>()
+                            t => t.GetCustomAttribute<DiscordDefaultMemberPermissionsAttribute>()
                         )
-                        .ToList();
+                        .ToArray();
 
-                    if (defaultPermissionAttributes.Count > 1)
+                    var directMessagePermisisonAttributes = groupNode.GroupTypes.Select
```

Comment:
Remember that if no attribute exists, this array will contain nulls and be > 1. Filter them out first.